test/e2e/networking.go

```diff
@@ -71,140 +70,198 @@ var _ = Describe("Networking", func() {
 		}
 	})
 
-	//Now we can proceed with the test.
-	It("should function for intra-pod communication", func() {
-		if testContext.Provider == "vagrant" {
-			By("Skipping test which is broken for vagrant (See https://github.com/GoogleCloudPlatform/kubernetes/issues/3580)")
-			return
-		}
+	//Two parallel array - services[0] corresponds to timeouts[0]
+	services := [...]int{1, 2}         // the first test is equivalent to the original networking.go
+	timeouts := [...]float64{300, 600} // very liberal timeout policies.  Should be more like 100 seconds.
 
```

Comment:
If these arrays are linked you should do

    []struct{
      service int
      timeout float64 # (should really be time.Duration)
    }{
      {1, 5 * time.Minute},
      ...
    }